schulcloud/Controllers/Dashboard/HomeworkOverviewViewController.swift

```diff
@@ -63,6 +64,13 @@ final class HomeworkOverviewViewController: UIViewController {
 
     @objc func updateHomeworkCount() {
         let fetchedObject = (self.resultController.fetchedObjects ?? []) as [Homework]
+        if fetchedObject.count > 0 {
+            self.showAllHomework.isEnabled = true
+            self.showAllHomework.setTitle("Show All(\(fetchedObject.count))", for: .normal)
+        } else {
+            self.showAllHomework.isEnabled = false
```

Comment:
Here you have to set the title as well.
I also would prefer such code
```swift
self.showAllHomework.isEnabled = fetchedObject.isEmpty ? fasle : true
// or even shorter
self.showAllHomework.isEnabled = !fetchedObject.isEmpty
```
over
```swift
if fetchedObject.count > 0 {
  self.showAllHomework.isEnabled = true
} else {
  self.showAllHomework.isEnabled = false
}
```
I think it's called result-driven vs input-driven programming. I'm for result-driven because you can easily forget to set an attribute. Like here.

Also `!array.isEmpty` over checking `arrat.count > 0`